metacat-common-server/src/main/java/com/netflix/metacat/common/server/properties/Config.java

```diff
@@ -266,4 +266,10 @@
      * @return The partition topic ARN or null if no property set
      */
     String getSnsTopicPartitionArn();
+
+    /**
+     * Use default user metadata service.
+     * @return The flag to turn on mysql based user metadata service.
+     */
+    boolean userMetadataServiceEnabled();
```

Comment:
a comment here about the general idea. there are two ways to do this.

1) Disable the feature completely. So that there is no trace of it in the system at runtime. When it is enabled specify which implementation to use somehow (profile, property, etc)
2) Enable it by default but switch the implementation based on profile, property, etc

This looks like number 2 but the naming of this variable is distinctly pointing towards number 1. I believe we discussed that unfortunately the user metadata service is pretty hardcoded into the application all over the place so 1 may be hard to accomplish but perhaps we should think of doing it.

One way I could see doing it is abstract the entire user metadata behind an adapter. Or the default implementation could really be the adapter. You can give it an optional constructor argument of another implementation and then if that exists at runtime defer to it or if it doesn't return nothing. That way you can still keep the various points in the code where the user metadata service is being called but be able to turn on or off a real implementation. Not sure how others feel or if this is clear.